mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -339,7 +340,17 @@ private boolean identicalFlags(ComposedMessageIdWithMetaData oldComposedId, Flag
     }
 
     private Mono<Pair<Flags, ComposedMessageIdWithMetaData>> updateFlags(ComposedMessageIdWithMetaData oldComposedId, ComposedMessageIdWithMetaData newComposedId) {
-        return imapUidDAO.updateMetadata(newComposedId, oldComposedId.getModSeq())
+        ComposedMessageId composedMessageId = oldComposedId.getComposedMessageId();
```

Comment:
Should it take the composedMessageId from `newComposedId` here instead of the old one?